src/python/pants/core/util_rules/strip_source_roots.py

```diff
@@ -30,6 +33,27 @@ class SourceRootStrippedSources:
     """Wrapper for a snapshot of files whose source roots have been stripped."""
 
     snapshot: Snapshot
+    # source root -> file paths relative to that root.
+    # Note that this will not contain entries for files to which the concept of source roots
+    # doesn't apply (e.g., the sources of a files(...) target), even when such files are
+    # in the snapshot.
+    root_to_relfiles: FrozenDict[str, Tuple[str, ...]]
+
+    @classmethod
+    def for_single_source_root(
+        cls, snapshot: Snapshot, source_root: str
+    ) -> "SourceRootStrippedSources":
+        return cls(snapshot, FrozenDict({source_root: snapshot.files}))
+
+    def get_file_to_stripped_file_mapping(self) -> FrozenDict[str, str]:
```

Comment:
Great idea. Maybe make this memoized to avoid continually iterating over the collection?

Reply:
Also, this should probably be a property called `file_to_stripped_file_mapping`?

Reply:
I'd rather keep a clear distinction between the @dataclass properties and computed data such as this.